### DISCUSSION TITLE
Property 'includes' of object is not a function

### DISCUSSION THREAD

**Discussion by croban** [sebastienros/jint#1585] [Q&A] [answered]
Inside 3.0.0-beta-2049 Version 

The code below throws Exception `Jint.Runtime.JavaScriptException : Property 'includes' of object is not a function`

```
            var employees = new List<string>{ "test1", "test2" };

            var engine = new Engine(options => {
            });

            engine.Execute(@"function myFunction(employees) {
               return employees.includes('test1');
            }");
            var updatedEmployees = engine.Invoke("myFunction", JsValue.FromObject(engine, employees)).ToObject() as bool?;
            
            Assert.IsTrue(updatedEmployees);
````

**Expected behavior**

CLR Type `List<string>` should be converted to JS Array or we have to manually specify converter for List<T>?

**Comment by lahma** [comment, marked as answer]
Jint only automatically handles arrays to JS array conversion as it's safe as being fixed size. Either call ToArray() when setting value to engine or modify options object wrapped to set different prototype.

https://github.com/sebastienros/jint/blob/db73cadf0a05d2485de49380a06bf7fea5b4f3b7/Jint.Tests/Runtime/InteropTests.cs#L848-L879

**Reply by croban** [reply]
Okay understand, I've helped my self with this:

```c#
public class ListConverter : IObjectConverter
{
    public bool TryConvert(Engine engine, object value, out JsValue? result)
    {
        if (value.GetType().IsGenericType && value is IList temp)
        {
            object[]? arr = new object[temp.Count];
            temp.CopyTo(arr, 0);
            result = JsValue.FromObject(engine, arr);
            return true;
        }

        result = null;
        return false;
    }
}
```

**Reply by croban** [reply]
Your solution is pretty slick! I'm still getting the hang of the library, so I need to delve deeper. The SetWrapObjectHandler function seems like a bit of magic to me.
